spyder/app/tests/test_mainwindow.py

```diff
@@ -2421,6 +2421,60 @@ def test_preferences_change_font_regression(main_window, qtbot):
                     timeout=5000)
 
 
+@pytest.mark.slow
+@pytest.mark.skipif(
+    sys.platform == 'darwin',
+    reason="Changes of Shitf+Return shortcut cause an ambiguos shortcut")
+def test_preferences_empty_shortcut_regression(main_window, qtbot):
+    """
+    Test for spyder-ide/spyder/#12992 regression.
+
+    Overwritting shortcuts results in a shortcuts conflict.
+    """
+    # Wait until the window is fully up
+    shell = main_window.ipyconsole.get_current_shellwidget()
+    qtbot.waitUntil(lambda: shell._prompt_html is not None,
+                    timeout=SHELL_TIMEOUT)
+
+    # Setup shortcuts (set run cell and advance shortcut to run selection)
+    base_run_cell_advance = CONF.get_shortcut(
+        'editor', 'run cell and advance')  # Should be Shift+Return
+    base_run_selection = CONF.get_shortcut(
+        'editor', 'run selection')  # Should be F9
+    assert base_run_cell_advance == 'Shift+Return'
+    assert base_run_selection == 'F9'
+    CONF.set_shortcut(
+        'editor', 'run cell and advance', '')
+    CONF.set_shortcut(
+        'editor', 'run selection', base_run_cell_advance)
+    main_window.apply_shortcuts()
+
+    # Check execution of shortcut
+    # Create new file
+    main_window.editor.new()
+    code_editor = main_window.editor.get_focus_widget()
+    code_editor.set_text('print(0)\nprint(1)\nprint(2)')
+
+    with qtbot.waitSignal(shell.executed):
+        qtbot.keyClick(code_editor, Qt.Key_Return, modifier=Qt.ShiftModifier)
+    qtbot.waitUntil(lambda: 'print(0)' in shell._control.toPlainText())
+    assert 'print(1)' not in shell._control.toPlainText()
```

Comment:
I think this is incorrect because this is what you get when you have the following code:

```python
print(0)
print(1)
print(2)
```

and press `Shift+Enter` (i.e. in case the shortcut was not changed correctly):

![Selección_013](https://user-images.githubusercontent.com/365293/85318593-df67e700-b485-11ea-9595-ce29cd694ba9.png)

As you can see, there's no `print(1)` there.

Reply:
So I'd suggest to change the text to be executed to

    code_editor.set_text("print(0)\nprint('ññ')")

and assert that `ññ` is not present in `shell._control.toPlainText()`. That's because there are lot of other numbers in the console (for instance in the banner on the top) that could match other print's with numbers.